### PR TITLE
1. Added a functioning landing page 2. Changed Green circle to cash app logo 3. Added animation effects for navigation bar 4. Also added animation effects for " text " and "image " to fade in - but not functioning yet

### PR DESCRIPTION
**Pull request by aainagargg**
Particular details mentioned in the commit message.

### CODE REVIEW

**Comment by SaahilKapasi** [comment]
It'd be great if you modify this so that it works with the latest backend code, currently this frontend code is still looking for fprVarianceScore and fprMeanScore when only fprScore exists now.

**Comment by YesIfwRONG** [comment]
> It'd be great if you modify this so that it works with the latest backend code, currently this frontend code is still looking for fprVarianceScore and fprMeanScore when only fprScore exists now.

In my branch, I changed the frontend to look for fprScore instead. As long as we merge both mine and hers, things should work, if nothing unexpected happens.

**Review by YesIfwRONG — APPROVED** [review]

**Comment by SaahilKapasi** [comment]
> > It'd be great if you modify this so that it works with the latest backend code, currently this frontend code is still looking for fprVarianceScore and fprMeanScore when only fprScore exists now.
> 
> In my branch, I changed the frontend to look for fprScore instead. As long as we merge both mine and hers, things should work, if nothing unexpected happens.

If possible I'd like to see a deployed version before merging anything. The merge conflicts also have to be resolved, could you try merging your branch into hers or vice versa?

**Comment by aainagargg** [comment]
@YesIfwRONG are you working on merging, refactoring this then?

File: view/src/pages/LandingPage.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
```
Comment (CodeYimin) [inline view/src/pages/LandingPage.tsx:1]: There seems to be a landing page here, but also one in App.tsx

File: view/src/App.tsx [hunk 2]
```diff
@@ -1,86 +1,148 @@
...
+import React, { useState, useEffect } from "react";
 import Navbar from "./components/Navbar";
-import { API_BASE_URL } from "./consts/consts";
-import GraphPage from "./pages/GraphPage";
-import WelcomePage from "./pages/WelcomePage";
 import { Dataset } from "./types/types";
+import logo from "./logo.svg"; // Replace with your actual image path
 
 function App() {
-  const [page, setPage] = useState<"welcome" | "graph">("welcome");
```
Comment (CodeYimin) [inline view/src/App.tsx:10]: The entire main functionality of the app has been deleted here and has not been rewritten

File: view/src/App.tsx [hunk 3]
```diff
@@ -1,86 +1,148 @@
...
+                                    <li>Install dependencies for both backend and frontend.</li>
+                                    <li>Run the backend server using FastAPI.</li>
+                                    <li>Run the frontend development server using npm.</li>
+                                </ol>
+                            </div>
+                        </section>
+                    </>
+                ) : page === "welcome" ? (
+                    /* Welcome Page */
+                    <>Welcome Page Content Here</> // Replace with your actual WelcomePage component
```
Comment (CodeYimin) [inline view/src/App.tsx:136]: The actual WelcomePage component already existed in the previous code, there is no need to replace the actual code with a placeholder

File: view/src/App.tsx [hunk 4]
```diff
@@ -1,86 +1,148 @@
...
+                ) : page === "graph" && selectedDataset ? (
+                    /* Graph Page */
+                    <>Graph Page Content Here</> // Replace with your actual GraphPage component
```
Comment (CodeYimin) [inline view/src/App.tsx:139]: ^^ Same with graph page

File: view/src/App.tsx [hunk 5]
```diff
@@ -1,86 +1,148 @@
...
+        <div className="min-h-screen">
+            <Navbar
+                datasets={datasets}
+                selectedDataset={selectedDataset}
+                onSelectDataset={(dataset) => setSelectedDataset(dataset)}
+                onNewDataset={handleNewDataset}
+                isLandingPage={page === "landing"}
+            />
+
+            {/* Main Content Area */}
```
Comment (CodeYimin) [inline view/src/App.tsx:35]: This seems to be a landing page, but there's another landing page in LandingPage.tsx

File: view/src/components/Navbar.tsx [hunk 6]
```diff
@@ -3,60 +3,73 @@ import { Dataset } from "../types/types";
...
-  datasets,
-}: NavbarProps): ReactElement {
-  const [modalOpen, setModalOpen] = useState(false);
+                                   onSelectDataset,
+                                   selectedDataset,
+                                   onNewDataset,
+                                   datasets,
+                                   isLandingPage = false, // Default to false
+                               }: NavbarProps): ReactElement {
+    const [modalOpen, setModalOpen] = useState(false);
```
Comment (CodeYimin) [inline view/src/components/Navbar.tsx:20]: There should not be this many tabs, the previous tabbing was fine

File: view/src/components/Navbar.tsx [hunk 7]
```diff
@@ -3,60 +3,73 @@ import { Dataset } from "../types/types";
...
-      <div className="flex items-center">
-        {/* Placeholder for logo- added */}
-        <div className="h-8 w-8 bg-green-500 rounded-full mr-3"></div>
-        <span className="text-xl font-semibold text-gray-800">Cash App</span>
-      </div>
+    return (
+        <div className="fixed top-0 left-0 w-full z-50 flex justify-between items-center p-4 bg-white shadow-md">
+            {/* Left Section: Cash App Logo */}
+            <div className="flex items-center">
+                <img
```
Comment (CodeYimin) [inline view/src/components/Navbar.tsx:26]: We should be using 2 space tabbing

**Review by CodeYimin — CHANGES_REQUESTED** [review]
The preview deployment does not run.

All the previous code such as loading indicators would be gone, the pages we had wouldn't load etc
A landing page component was created but then didn't use it and instead wrote directly inside the main file.

This can't be merged since it completely breaks what we had working before

**Comment by CodeYimin** [comment]
I'd recommend putting the landing page inside LandingPage.tsx and keeping App.tsx mostly intact to avoid merge conflicts

**Comment by aainagargg** [comment]
Addressed these issues by merging into Yimin's code.

**Pull request closed without merging**